### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by soniaroh**
wrote some tests, still in progress to pass them all
modified code with ES-6
added half-star maker function
added product id number to pathname

### CODE REVIEW

**Comment by avincenthill** [comment]
Better commit messages, still some to be done in imperative tense next time. Also CircleCI build failing due to nonexistant id array in your testing suite

File: .env [hunk 1]
```diff
@@ -1,3 +1,3 @@
 DB_HOST=localhost
 DB_USER=root
-DB_PASS='password'
\ No newline at end of file
+DB_PASS='password'
```
Comment (avincenthill) [inline .env]: this file needs to stop being committed Sonia

File: .vscode/settings.json [hunk 2]
```diff
@@ -1,2 +1,5 @@
 {
+    "editor.tabSize": 2,
+    "editor.formatOnSave ": true,
+    "editor.insertSpaces": false,
```
Comment (avincenthill) [inline .vscode/settings.json:4]: Probably best to have vscode settings in containing folders of your projects unless this is project specific settings

File: client/src/App.jsx [hunk 3]
```diff
@@ -1,20 +1,11 @@
...
+const App = () => (
+  <div>
+    <RelatedItems />
+  </div>
+);
 
-class App extends React.Component {
-	render() {
-		return (
-			<div>
```
Comment (avincenthill) [inline client/src/App.jsx:13]: Nice cleanup here

File: database/index.js [hunk 4]
```diff
@@ -1,68 +1,82 @@
...
+const mysql = require('mysql');
+const faker = require('faker');
 
-
-let connection = mysql.createConnection({
-    host: 'localhost',
-    user: 'root',
-    password: 'password',
-    database: 'slideShowData'
+// tried using process.env.PASSWORD but made database unreachable
```
Comment (avincenthill) [inline database/index.js]: fix this, can't have passwords hardcoded into the product

File: server/index.js [hunk 5]
```diff
@@ -1,41 +1,41 @@
...
+      return res.json({
+        data: results,
+      });
+    }
+  });
 });
 
 
-const PORT = 4043;
+const PORT = process.env.url || 4043;
```
Comment (avincenthill) [inline server/index.js:30]: confusing name for port var here

**Review by avincenthill — CHANGES_REQUESTED** [review]

File: .env (hunk 1 above)
Comment (michchen58) [inline .env]: should we avoid committing passwords?

File: client/src/PzElement.jsx [hunk 6]
```diff
@@ -1,14 +1,73 @@
...
+// =============================================
+
+// function shortens product discription:
+
+const lessWord = str => ((str.length > 50) ? str.split('').slice(0, 50).join('') : str);
+
+// =============================================
+
+
+const PzElement = props => (
```
Comment (michchen58) [inline client/src/PzElement.jsx:52]: What does Pz mean?

File: client/src/PzElement.jsx [hunk 7]
```diff
@@ -1,14 +1,73 @@
...
+  <div className={styles.child}>
+    <img alt="product" width="160 vmin" height="160 vmin" className={styles.element} src={props.element.imageURL} />
+    <div className={styles.text}>
+      {lessWord(props.element.productDescription)}
```
Comment (michchen58) [inline client/src/PzElement.jsx:56]: ESLint will tell you to deconstruct props and styles

**Review by michchen58 — APPROVED** [review]

File: .env (hunk 1 above)
Comment (mattminwoolee) [reply .env]: use .gitignore

File: client/src/tests/__snapshots__/App.test.js.snap [hunk 8]
```diff
@@ -0,0 +1,19 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
```
Comment (mattminwoolee) [inline client/src/tests/__snapshots__/App.test.js.snap:1]: maybe you can put these into a .gitignore file

**Review by mattminwoolee — APPROVED** [review]

**Pull request merged**